.github/workflows/build.yml

```diff
@@ -0,0 +1,116 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+      - gh-actions # TODO: remove this.
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: '0 4 * * *' # Runs every day at 4am: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule
+
+jobs:
+
+  build:
+    name: ${{ matrix.os }}, Node.js v${{ matrix.node }}, Python ${{ matrix.python }}
+
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: ['12.x']
+        python: ['2.x']
+        include:
+          - os: ubuntu-latest
+            node: '12.x'
+            python: '3.x'
+            tests: 'skip'
+
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```

Comment:
may probably require `fetch-depth: "0"`

Reply:
Let's do that. I do not have any objections. Usually, you use `fetch-depth: '0'` if you want to check existing tags/releases in the repo and use it in your step/job condition. Otherwise, the default `1` is faster.

As I started in the PR description, this will be most likely required to make the yarn caching with lerna run properly with GH Actions.

> * It does not cache things for `yarn` and `node_modules` (with `lerna`, but someone had a similar issue, and was solved like [this](https://github.com/markuplint/markuplint/pull/111/files)).

I am going to make the suggested change.

Reply:
@benoitf, out of curiosity, did you intentionally mention only this `checkout` action and not the other for the npm publish build? Just asking, though.

Reply:
@kittaakos in fact yes it was more for the 'build' usecase